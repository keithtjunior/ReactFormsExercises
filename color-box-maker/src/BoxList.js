import React, {useState} from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';

const BoxList = () => {
    const INITIAL_STATE = []
    const [boxes, setBoxes] = useState(INITIAL_STATE);
    
    const createBox = (box) => {
        setBoxes(boxes => [...boxes, box])
    }
    const deleteBox = (id) => {
        const newBoxes = [...boxes];
        newBoxes.splice(newBoxes.findIndex(box => id === box.id), 1);
        setBoxes(newBoxes);
    }

    return (
        <div>
            <NewBoxForm createBox={createBox}/>
            {boxes.map(box => (
                <Box 
                key={box.id}
                id={box.id}
                width={box.width} 
                height={box.height} 
                color={box.color}
                deleteBox={deleteBox}/>
            ))}
        </div>
    )
}

export default BoxList;