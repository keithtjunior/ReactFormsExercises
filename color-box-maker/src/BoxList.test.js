import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function() {
  render(<BoxList />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can create a new box", function() {
    const { getByLabelText, queryByText } = render(<BoxList />);
  
    expect(queryByText("X")).not.toBeInTheDocument();
  
    const nameInput = getByLabelText("Width");
    const qtyInput = getByLabelText("Height");
    const colorInput = getByLabelText("Color");
    const submitBtn = queryByText("Create Box")
  
    // fill out the form
    fireEvent.change(nameInput, { target: { value: 200 }});
    fireEvent.change(qtyInput, { target: { value: 200 }});
    fireEvent.change(colorInput, { target: { value: 'blue' }});
    fireEvent.click(submitBtn);
  
    expect(queryByText("X")).toBeInTheDocument();
});