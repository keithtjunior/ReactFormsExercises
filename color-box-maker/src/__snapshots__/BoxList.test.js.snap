// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="margin: 2rem;"
    >
      <form>
        <div>
          <label
            for="width"
          >
            Width: 
          </label>
          <input
            id="width"
            max="1420"
            min="1"
            name="width"
            required=""
            style="margin-top: 10px;"
            type="number"
            value="1"
          />
        </div>
        <div>
          <label
            for="height"
          >
            Height: 
          </label>
          <input
            id="height"
            max="1420"
            min="1"
            name="height"
            required=""
            style="margin-top: 10px;"
            type="number"
            value="1"
          />
        </div>
        <div>
          <label
            for="color"
          >
            Color: 
          </label>
          <input
            id="color"
            maxlength="25"
            minlength="1"
            name="color"
            placeholder="Enter color of box"
            required=""
            style="margin-top: 10px;"
            type="text"
            value=""
          />
        </div>
        <button
          style="margin-top: 10px;"
        >
          Create Box
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
